### ISSUE TITLE
Will not work if elements have different height

### ISSUE THREAD

**Issue by nikkanetiya** [paulcollett/vue-masonry-css#2]
Hello,

I have created something same for one of my project, I was checking your repo. and by seeing your code I found that it will not work if elements have different height because we are counting elements in each column. Any plans to support variable heights?

**Comment by paulcollett** [comment]
If I follow correctly, smart sorting, based on height, to display as close to the original order would be a great feature! Is that what your suggesting?

We do enter the territory of having to pre-render elements for height detection, so I'd be thinking this would be a nice optional feature.

**Comment by vitorleonel** [comment]
Hi,

I applied on my portal and liked a results with elements have different height.

See on https://www.treinarfrontend.com/

**Comment by paulcollett** [comment]
@vitorleonel, looks like smart sorting based on heights would help you here too. It would help balance all the items and get rid of that long first column that I see. I'll put some thought into this

**Comment by nikkanetiya** [comment]
@vitorleonel, Looks pretty fine in your case, As per my previous short experience, it was causing the issue, Will check it again and prepare the fiddle if it's not working properly when I can.

**Comment by batraabhishek** [comment]
Hi, @paulcollett Awesome work with this Vue Masonry Css. I am facing a similar issue. I have dynamic content with different height. Is there any way to balance the columns by height, instead of a number of elements in column.  

PS: I am still learning Web development and frontend technologies.

**Comment by helariL** [comment]
@paulcollett any updates on this? It would perfect to celebrate issue birthday with implement of it 😄

**Comment by graphek** [comment]
joining the party here - i really would appreciate a solution, with variable height items, since one columns always gets a lot longer than the other =)

**Comment by gokhancetiner** [comment]
Hello @paulcollett ,

Is it possible to order elements with column heights instead of the number of elements? Because some of the height of the column is much longer than the others. 

Many  Thanks

**Comment by liho98** [comment]
Yeah, this feature i think is quite essential for vertical masonry
